modules/local/busco.nf

```diff
@@ -36,50 +40,70 @@ process BUSCO {
     else
         cp_augustus_config = "N"
 
-    """
-    # get path to custom config file for busco (already configured during conda installation)
-    busco_path="\$(which busco)"
-    config_file="\${busco_path%bin/busco}share/busco/config.ini"
+    if (params.busco_reference)
+        lineage_dataset_provided = "Y"
+    else
+        lineage_dataset_provided = "N"
 
+    if (params.busco_reference){
+        p = "--lineage_dataset dataset/${db}"
+    } else {
+        if (params.busco_auto_lineage_prok)
+            p = "--auto-lineage-prok"
+        else
+            p = "--auto-lineage"
+        if (params.busco_download_path)
+            p += " --offline --download_path ${download_folder}"
+    }
+    """
     # ensure augustus has write access to config directory
     if [ ${cp_augustus_config} = "Y" ] ; then
         cp -r /usr/local/config/ augustus_config/
         export AUGUSTUS_CONFIG_PATH=augustus_config
     fi
 
     # place db in extra folder to ensure BUSCO recognizes it as path (instead of downloading it)
-    mkdir dataset
-    mv ${db} dataset/
+    if [ ${lineage_dataset_provided} = "Y" ] ; then
+        mkdir dataset
+        mv ${db} dataset/
+    fi
 
-    busco --lineage_dataset dataset/${db} \
+    if busco ${p} \
         --mode genome \
         --in ${bin} \
-        --config \${config_file} \
         --cpu "${task.cpus}" \
-        --out "BUSCO" > ${bin}_busco.log
-
-    # get used db name
-    # (set nullgob: if pattern matches no files, expand to a null string rather than to itself)
-    shopt -s nullglob
-    summaries=(BUSCO/short_summary.specific.*.BUSCO.txt)
-    if [ \${#summaries[@]} -ne 1 ]; then
-        echo "ERROR: none or multiple 'BUSCO/short_summary.specific.*.BUSCO.txt' files found. Expected one."
+        --out "BUSCO" > ${bin}_busco.log 2> ${bin}_busco.err; then
+
+        # get used db name
+        # (set nullgob: if pattern matches no files, expand to a null string rather than to itself)
+        shopt -s nullglob
+        summaries=(BUSCO/short_summary.specific.*.BUSCO.txt)
+        if [ \${#summaries[@]} -ne 1 ]; then
+            echo "ERROR: none or multiple 'BUSCO/short_summary.specific.*.BUSCO.txt' files found. Expected one."
+            exit 1
+        fi
```

Comment:
Curious: in what case can there be several specific summary files?

Reply:
I don't know :D I think I just wanted to check if it is exactly 1